src/devices/src/virtio/net/device.rs

```diff
@@ -679,12 +679,8 @@ impl VirtioDevice for Net {
         self.device_activated
     }
 
-    fn set_device_activated(&mut self, device_activated: bool) {
-        self.device_activated = device_activated;
-    }
-
-    fn activate(&mut self, _mem: GuestMemoryMmap) -> ActivateResult {
-        // TODO: to be removed
+    fn activate(&mut self) -> ActivateResult {
```

Comment:
How does this play out with resetting the device? I am not very sure, but I think we should be able to say that the device is no longer active in case of error, which implies this function should receive a parameter for setting the activate on true or false. I can't find the `reset` code in the code base. Was it removed?

Reply:
So none of our devices support `reset` so they just inherit the default trait implementation:
```rust
    /// Optionally deactivates this device and returns ownership of the guest memory map, interrupt
    /// event, and queue events.
    fn reset(&mut self) -> Option<(EventFd, Vec<EventFd>)> {
        None
    }
```

This is used in `src/devices/src/virtio/mmio.rs:194` when the driver requests a reset:
```rust
            _ if status == 0 => {
                if self.locked_device().is_activated() {
                    let mut device_status = self.device_status;
                    let reset_result = self.locked_device().reset();
                    match reset_result {
                        Some((_interrupt_evt, mut _queue_evts)) => {}
                        None => {
                            device_status |= FAILED;
                        }
                    }
                    self.device_status = device_status;
                }

                // If the backend device driver doesn't support reset,
                // just leave the device marked as FAILED.
                if self.device_status & FAILED == 0 {
                    self.reset();
                }
            }
```

So we just mark our devices as `FAILED`.

Reply:
Ok, we can add a parameter to `activate` when we implement `reset`.